packages/web/src/generic.ts

```diff
@@ -11,6 +11,8 @@ export const inject = (
 
   const mode = getMode(props.mode);
 
+  window.vam = mode;
```

Comment:
Let's create `setMode` which does set `window.vam`

I would suggest that we create `isProduction` and `isDevelopment` – both are reading from `window.vam` so we're consistent every time we try to identify the correct environment.

This should also eliminate the need of manually reading `window.vam` on `track`

Reply:
I renamed `isDevelopment` to `detectDevelopment` and `getMode` to `detectMode` 
`setMode` and `getMode` are now just wrapping `window.vam` and `isProduction` and `isDevelopment` is reading from `getMode`